onnxruntime/core/providers/cpu/nn/pool_base.h

```diff
@@ -140,18 +148,22 @@ class PoolBase {
       }
 
       ORT_ENFORCE(strides_.size() == kernel_shape_.size());
+      ORT_ENFORCE(dilations_.size() == kernel_shape_.size(),
+                  "Dilations dimensions should match kernel shape");
     }
   }
 
   ~PoolBase(){};
 
   std::vector<int64_t> SetOutputSize(const TensorShape& input_shape,
                                      int64_t output_channel,
-                                     std::vector<int64_t>* pads) const {
+                                     std::vector<int64_t>* pads,
```

Comment:
>std::vector<int64_t>* pads [](start = 37, length = 26)

Is pads optional argument? Should be a reference not a pointer

Reply:
This is legacy -- not sure about the downstream impact. Will leave as-is for now.